### ISSUE TITLE
Task_3_Churn_Prediction.ipynb

### ISSUE THREAD

**Issue by noormaryamqurexhi-spec** [noormaryamqurexhi-spec/DevelopersHub-Internship-Tasks#3]
# Task 3: Customer Churn Prediction (Bank Customers)
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import confusion_matrix, classification_report

# 1. Load the Dataset
# (Using a standard link to a churn dataset so the code runs immediately)
url = 'https://raw.githubusercontent.com/afrizal423/Customer-Churn-Prediction/master/data/Churn_Modelling.csv'
df = pd.read_csv(url)

# 2. Data Cleaning & Preparation
# Removing columns that don't help with prediction (Names, IDs)
df = df.drop(['RowNumber', 'CustomerId', 'Surname'], axis=1)

# 3. Encoding Categorical Features (Converting text to numbers)
le = LabelEncoder()
df['Geography'] = le.fit_transform(df['Geography'])
df['Gender'] = le.fit_transform(df['Gender'])

# 4. Splitting Data
X = df.drop('Exited', axis=1) # Features
y = df['Exited']              # Target (1 = Left, 0 = Stayed)

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# 5. Training the Model
model = RandomForestClassifier(n_estimators=100, random_state=42)
model.fit(X_train, y_train)

# 6. Feature Importance Visualization
# This shows what influences a customer to leave the most
plt.figure(figsize=(10,6))
feat_importances = pd.Series(model.feature_importances_, index=X.columns)
feat_importances.nlargest(10).plot(kind='barh', color='skyblue')
plt.title('Top Factors Influencing Customer Churn')
plt.xlabel('Importance Score')
plt.show()

# 7. Model Evaluation
predictions = model.predict(X_test)
print("--- Classification Report ---")
print(classification_report(y_test, predictions))
